### PR TITLE
fix(DB/creature) add missing stealth detection to mobs

### PR DESCRIPTION
**Pull request by v-mstrs**
## Changes Proposed:
-  add missing stealth detection to mobs

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/16221
- Closes https://github.com/chromiecraft/chromiecraft/issues/5645
- Closes https://github.com/chromiecraft/chromiecraft/issues/5580

## Tests Performed:

- Tested

## How to Test the Changes:

While stealthed (.aura 1784)

1. .go cr 71514  

1. .tele kara 
2. Walk next to the sentries/valets.
3. Observe they do see you.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1684589406305000945.sql [hunk 1]
```diff
@@ -0,0 +1,10 @@
+--
+
+UPDATE `creature_addon` SET `auras` = '18950' WHERE `guid` IN (135170, 135171, 135172, 135173, 135174, 135175, 135345, 135346, 135347, 135348, 135349, 135350, 135351, 135352, 135353, 135354, 135355, 135356, 135357, 135358, 135359, 135360, 135361, 135362);
+INSERT INTO `creature_addon` (`entry`, `path_id`, `mount`, `bytes1`, `bytes2`, `emote`, `visibilityDistanceType`, `auras`) VALUES
+(16408, 0, 0, 0, 0, 0, 0, '18950');
+INSERT INTO `creature_template_addon` (`entry`, `path_id`, `mount`, `bytes1`, `bytes2`, `emote`, `visibilityDistanceType`, `auras`) VALUES
+(16408, 0, 0, 0, 0, 0, 0, '18950');
+
+UPDATE `creature_addon` SET `auras` = 18950  WHERE `guid` in (20058, 16424)
```
Comment (Gultask) [inline data/sql/updates/pending_db_world/rev_1684589406305000945.sql:9]: This isn't needed as it'd affect unintended creatures. creature_addon is tied to guid (an unique spawn in the world) while creature_template_addon is tied to entry (the template of the creature.
Also the field is a string, not an int :P 
```suggestion
```

File: data/sql/updates/pending_db_world/rev_1684589406305000945.sql [hunk 2]
```diff
@@ -0,0 +1,10 @@
+--
+
+UPDATE `creature_addon` SET `auras` = '18950' WHERE `guid` IN (135170, 135171, 135172, 135173, 135174, 135175, 135345, 135346, 135347, 135348, 135349, 135350, 135351, 135352, 135353, 135354, 135355, 135356, 135357, 135358, 135359, 135360, 135361, 135362);
+INSERT INTO `creature_addon` (`entry`, `path_id`, `mount`, `bytes1`, `bytes2`, `emote`, `visibilityDistanceType`, `auras`) VALUES
+(16408, 0, 0, 0, 0, 0, 0, '18950');
+INSERT INTO `creature_template_addon` (`entry`, `path_id`, `mount`, `bytes1`, `bytes2`, `emote`, `visibilityDistanceType`, `auras`) VALUES
+(16408, 0, 0, 0, 0, 0, 0, '18950');
```
Comment (Gultask) [inline data/sql/updates/pending_db_world/rev_1684589406305000945.sql:7]: Both these INSERTS require a DELETE preceding it. And it looks like the first one is like I said below

```suggestion

DELETE FROM `creature_template_addon` WHERE `entry` = 16408;
INSERT INTO `creature_template_addon` (`entry`, `path_id`, `mount`, `bytes1`, `bytes2`, `emote`, `visibilityDistanceType`, `auras`) VALUES
(16408, 0, 0, 0, 0, 0, 0, '18950');
```

File: data/sql/updates/pending_db_world/rev_1684589406305000945.sql [hunk 3]
```diff
@@ -0,0 +1,10 @@
+--
+
+UPDATE `creature_addon` SET `auras` = '18950' WHERE `guid` IN (135170, 135171, 135172, 135173, 135174, 135175, 135345, 135346, 135347, 135348, 135349, 135350, 135351, 135352, 135353, 135354, 135355, 135356, 135357, 135358, 135359, 135360, 135361, 135362);
```
Comment (Gultask) [inline data/sql/updates/pending_db_world/rev_1684589406305000945.sql:3]: It looks like there are some mistakes here.
135173 is an OLDWorld Trigger
135172 is a Shade of Aran Teleport Center

**Review by Gultask — CHANGES_REQUESTED** [review]

File: data/sql/updates/pending_db_world/rev_1684589406305000945.sql (hunk 3 above)
Comment (Gultask) [reply data/sql/updates/pending_db_world/rev_1684589406305000945.sql:3]:
```
SELECT * FROM `creature` WHERE `guid` IN (135170, 135171, 135172, 135173, 135174, 135175, 135345, 135346, 135347, 135348, 135349, 135350, 135351, 135352, 135353, 135354, 135355, 135356, 135357, 135358, 135359, 135360, 135361, 135362);
```

**Comment by heyitsbench** [comment]
Wasn't mentioned in this PR or the original issue, but sniffs do confirm that the wolves use aura 18950.

File: data/sql/updates/pending_db_world/rev_1684589406305000945.sql (hunk 3 above)
Comment (v-mstrs) [reply data/sql/updates/pending_db_world/rev_1684589406305000945.sql:3]: They show up in Keira spawn addon of the selected creature

Comment (Gultask) [reply data/sql/updates/pending_db_world/rev_1684589406305000945.sql:3]: which creature?

Comment (v-mstrs) [reply data/sql/updates/pending_db_world/rev_1684589406305000945.sql:3]: oops no they dont

**Review by Gultask — APPROVED** [review]
ty for PR!

**Pull request merged**